File: 0977. Squares of a Sorted Array/Solution.cpp
#include <vector>

using std::vector;

class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> res(nums.size(), 0);
        int l = 0, r = nums.size() - 1;
        int i = nums.size() - 1;
        while (l <= r)
        {
            if (abs(nums[l]) < nums[r])
            {
                res[i] = nums[r] * nums[r];
                r--;
            }
            else
            {
                res[i] = nums[l] * nums[l];
                l++;
            }
            i--;
        }
        return res;
    }
};

File: 0977. Squares of a Sorted Array/Solution.py
class Solution:
    def sortedSquares(self, nums):
        """
        type nums: List[int]
        rtype: List[int]
        """
        res = [0] * len(nums)
        l, r = 0, len(nums) - 1,
        i = r
        while l <= r:
            if abs(nums[l]) < nums[r]:
                res[i] = nums[r] ** 2
                r -= 1
            else:
                res[i] = nums[l] ** 2
                l += 1
            i -= 1
        return res